test/navigation/imu_preprocessing.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <navigation/preprocess_imu.hpp>
+#include <utils/manual_time.hpp>
+
+namespace hyped::test {
+
+bool checkArrayEquality(core::ImuData &imu_data_a, core::ImuData &imu_data_b)
+{
+  if (imu_data_a.size() == imu_data_b.size()) {
+    for (std::size_t i; i < imu_data_a.size(); ++i) {
+      if (imu_data_a.at(i) == imu_data_b.at(i)) {
+        continue;
+      } else {
+        return false;
+      }
+    }
+  } else {
+    return false;
+  }
+  return true;
+}
+
+void test()
+{
+  utils::ManualTime manual_time;
+  core::Logger logger("test", core::LogLevel::kFatal, manual_time);
+  navigation::ImuPreprocessor imu_processer(logger);
+}
+
+TEST(Imu, equal_data)
+{
+  utils::ManualTime manual_time;
+  core::Logger logger("test", core::LogLevel::kFatal, manual_time);
+  navigation::ImuPreprocessor imu_processer(logger);
+  core::RawImuData data                          = {{1, 1, 1}};
```

Comment:
Are these all initialised with correct dimensions?

Reply:
Yeah it initialises an array with all equal elements to the size of the array